tools/SourceKit/tools/sourcekitd/bin/XPC/Service/XPCService.cpp

```diff
@@ -305,7 +314,10 @@ static void sourcekitdServer_event_handler(xpc_connection_t peer) {
     sourcekitdServer_peer_event_handler(peer, event);
   });
 
-  MainConnection = peer;
+  // Update the main connection
+  if (MainConnection)
+    xpc_release(MainConnection);
+  MainConnection = xpc_connection_t(xpc_retain(peer));
```

Comment:
The generally safe retain/release pattern, in case the 2 objects (new and old) happen to be the same, is to retain the new one before releasing the old one:

temp_object = retain(new)
release(old)
old = temp_object

Reply:
Ok, I'll fix it up shortly!